File: JavaScript/strategy-pattern/ModelDuck.js
import Duck from './Duck.js';
import Quack from './Quack.js';
import FlyNoWay from './FlyNoWay.js';

export default class MallardDuck extends Duck {
  constructor() {
    super();
    this.quackBehavior = new Quack();
    this.flyBehavior = new FlyNoWay();
  }

  display() {
    console.log("I'am a model duck");
  }
}

File: Ruby/strategy-pattern/model_duck.rb
# frozen_string_literal: true

require_relative './duck'
require_relative './quack'
require_relative './fly_no_way'

# ModelDuck Class
class ModelDuck < Duck
  def initialize
    super
    @quack_behavior = Quack.new
    @fly_behavior = FlyNoWay.new
  end

  def display
    puts "I'am a model duck"
  end
end
